src/sonic-yang-models/yang-models/sonic-feature.yang

```diff
@@ -0,0 +1,71 @@
+module sonic-feature{
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-feature";
+    prefix port;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "Feature Table yang Module for SONiC OS";
+
+    container sonic-feature {
+
+        container FEATURE {
+
+            description "feature table in config_db.json";
+
+            list FEATURE_LIST {
+
+                key "name";
+
+                leaf name {
+                    description "feature name in Feature table";
+                    type string {
+                        length 1..32;
+                    }
+                }
+
+                leaf state {
+                    description "state of the feature";
+                    type stypes:feature_state;
+                }
+
+                leaf auto_restart {
+                    type stypes:feature_state;
+
+                }
+
+                leaf has_timer {
+                    description "This configuration identicates if there is
+                                timer associated to this feature";
```

Comment:
what's the purpose of this timer? time to start the feature?

Reply:
Some the features have timer associated. `snmp` is an example. This attributed to indicate that field